### PR TITLE
Issue #8933: add handling of empty string in IllegalTypeCheck property

### LINKED ISSUE #8933: IllegalType: inability to set default value for property memberModifiers
Check documentation: https://checkstyle.org/config_coding.html#IllegalType

```

$ cat Test.java 
public class Test {
   private Optional<String> field1;
}

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="IllegalType">
            <property name="illegalClassNames" value="java.util.Optional" />
            <property name="tokens" value="VARIABLE_DEF" />
            <property name="memberModifiers" value=""/>
        </module>
   </module>
</module>

$ java -jar checkstyle-8.36.2-all.jar -c config.xml Test.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - cannot initialize module IllegalType - Cannot set property 'memberModifiers' to ''
  at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:482)
  at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
  at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:404)
  at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:331)
  at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:190)
  at com.puppycrawl.tools.checkstyle.Main.main(Main.java:125)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module IllegalType - Cannot set property 'memberModifiers' to ''
  at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:125)
  at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
  at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:477)
  ... 5 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Cannot set property 'memberModifiers' to ''
  at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:241)
  at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:194)
  at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:120)
  ... 7 more
Caused by: java.lang.reflect.InvocationTargetException
  at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2128)
  at org.apache.commons.beanutils.PropertyUtilsBean.setSimpleProperty(PropertyUtilsBean.java:2109)
  at org.apache.commons.beanutils.BeanUtilsBean.copyProperty(BeanUtilsBean.java:437)
  at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:231)
  ... 9 more
Caused by: java.lang.IllegalArgumentException: given name 
  at com.puppycrawl.tools.checkstyle.utils.TokenUtil.getTokenId(TokenUtil.java:174)
  at com.puppycrawl.tools.checkstyle.checks.coding.IllegalTypeCheck.setMemberModifiers(IllegalTypeCheck.java:828)
  ... 17 more
Checkstyle ends with 1 errors.
```

Expected: work as `memberModifiers` is not mentioned in config.

### PR DESCRIPTION
**Pull request by strkkk**
Closes #8933 
Default value is changed from null to empty list, setter is corrected to filter empty strings.

It seems that regression is not required.

### CODE REVIEW

**Comment by romani** [comment]
Github, rebase

**Comment by romani** [comment]
Github, rebase

File: src/xdocs/config_coding.xml [hunk 1]
```diff
@@ -2895,7 +2895,7 @@ public void myTest() {
               <td>subset of tokens
                 <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html">TokenTypes</a>
               </td>
-              <td><code>no tokens</code></td>
```
Comment (romani) [inline src/xdocs/config_coding.xml:2898]: can we remove support for no tokens from our validator?
to make sure we will not do this problem in design again.

```
src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
1140:                        result = "no tokens";

```

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/xdocs/config_coding.xml (hunk 1 above)
Comment (strkkk) [reply src/xdocs/config_coding.xml:2898]: removed

**Review by esilkensen — APPROVED** [review]

**Comment by strkkk** [comment]
github, rebase

**Comment by romani** [comment]
Github, rebase

**Review by romani — APPROVED** [review]
Ok to merge , when CI is green

**Pull request merged**